plugins/inputs/elasticsearch_query/elasticsearch_query_test.go

```diff
@@ -0,0 +1,684 @@
+		false,
+		false,
+		false,
+	},
+	{
+		"query 9 - invalid function",
+		esAggregation{
+			Index:           testindex,
+			MeasurementName: "measurement9",
+			MetricFields:    []string{"size"},
+			FilterQuery:     "downloads",
+			MetricFunction:  "average",
+			DateField:       "@timestamp",
+			QueryPeriod:     queryPeriod,
+			Tags:            []string{},
+		},
+		nil,
+		map[string]string{"size": "long"},
+		nil,
+		true,
+		false,
+		false,
+	},
+	{
+		"query 10 - non-existing metric field",
+		esAggregation{
+			Index:           testindex,
+			MeasurementName: "measurement10",
+			MetricFields:    []string{"none"},
+			DateField:       "@timestamp",
+			QueryPeriod:     queryPeriod,
+			Tags:            []string{},
+		},
+		nil,
+		map[string]string{},
+		nil,
+		false,
+		false,
+		false,
+	},
+	{
+		"query 11 - non-existing index field",
+		esAggregation{
+			Index:           "notanindex",
+			MeasurementName: "measurement11",
+			DateField:       "@timestamp",
+			QueryPeriod:     queryPeriod,
+			Tags:            []string{},
+		},
+		nil,
+		map[string]string{},
+		nil,
+		false,
+		false,
+		true,
+	},
+	{
+		"query 12 - non-existing timestamp field",
+		esAggregation{
+			Index:           testindex,
+			MeasurementName: "measurement12",
+			MetricFields:    []string{"size"},
+			MetricFunction:  "avg",
+			DateField:       "@notatimestamp",
+			QueryPeriod:     queryPeriod,
+			Tags:            []string{},
+		},
+		[]aggregationQueryData{
+			{
+				aggKey:   aggKey{measurement: "measurement12", name: "size_avg", function: "avg", field: "size"},
+				isParent: true,
+			},
+		},
+		map[string]string{"size": "long"},
+		nil,
+		false,
+		false,
+		false,
+	},
+	{
+		"query 13 - non-existing tag field",
+		esAggregation{
+			Index:             testindex,
+			MeasurementName:   "measurement13",
+			MetricFields:      []string{"size"},
+			MetricFunction:    "avg",
+			DateField:         "@timestamp",
+			QueryPeriod:       queryPeriod,
+			IncludeMissingTag: false,
+			Tags:              []string{"nothere"},
+		},
+		[]aggregationQueryData{
+			{
+				aggKey:   aggKey{measurement: "measurement13", name: "size_avg", function: "avg", field: "size"},
+				isParent: false,
+			},
+			{
+				aggKey:   aggKey{measurement: "measurement13", name: "nothere", function: "terms", field: "nothere"},
+				isParent: true,
+			},
+		},
+		map[string]string{"size": "long"},
+		nil,
+		false,
+		false,
+		false,
+	},
+}
+
+func setupIntegrationTest() {
+	type nginxlog struct {
+		IPaddress    string    `json:"IP"`
+		Timestamp    time.Time `json:"@timestamp"`
+		Method       string    `json:"method"`
+		URI          string    `json:"URI"`
+		Httpversion  string    `json:"http_version"`
+		Response     string    `json:"response"`
+		Size         float64   `json:"size"`
+		ResponseTime float64   `json:"response_time"`
+	}
+
+	err := e.connectToES()
+	if err != nil {
+		e.Log.Errorf("error setting up integration test: %s", err.Error())
+		return
+	}
```

Comment:
Please either simply return the error here and check it in the caller or pass the `testing.T` in here and use `require.NoError()` to fail hard. I prefer the former but I'm also OK with the latter.